Wrong number of arguments: call(foo,[])